### ISSUE TITLE
Commenting under another members comment

### ISSUE THREAD

**Issue by gregclemens** [ActiveForums/ActiveForums#270]
What I want to know is how in AF do I comment on a comment so that the comment shows up under the comment I'm commenting on and not at the bottom of the page of comments.  

When I hit the reply button for a specific comment I would expect that my comment would eventually show up under that comment.  Make sense?  Is there a setting?  I tried altering the Topics Display and Topic Display from default to TopicView but that didn't seem to change anything.

I believe what I'm describing must be do-able since that's how it works on any social media site.  Individuals are able to comment on other people's comments and have their comments show up under that person's comments, indented.  Maybe I'm missing something.

**Comment by supermansrf** [comment]
@gregclemens I was the one who replied to your email asking you to post this on here.  I didn't want to open dialog via email just to come on here and ask more questions.  

I'll be honest with you, the word comment was present 6 times in the first sentence.  You lost me at comment :smiley: 

I believe what you are talking about is commenting on a reply as opposed to using the quote feature?  I'm going to post the order these were entered on the forum at the beginning of the line like (1), (2)

(1)Topic:  Cutting down a tree
(1)Body:  I cut down a tree today

(2)Reply 1:  Was it easy?
(5)Comment to Reply 1: Not at all, almost killed myself
(7)Comment to Reply 1: Maybe next time you should hire a professional

(3)Reply 2:  Quit cutting down trees.  Love them instead
(4)Comment to Reply 2: You're a hippy, go eat some granola
(6)Comment to Reply 2: You got something against hippies you tree killer!?

-----------------End of comedic topic-----------------

On the forum currently it would look like this:

(1)Topic:  Cutting down a tree
(1)Body:  I cut down a tree today

(2)Reply 1:  Was it easy?
(3)Reply 2:  Quit cutting down trees.  Love them instead
(4)Comment to Reply 2: You're a hippy, go eat some granola
(5)Comment to Reply 1: Quote: Was it easy /Quote - Not at all, almost killed myself
(6)Comment to Reply 2: Quote: You're a hippy, go eat some granola /Quote - You got something against hippies you tree killer!?
(7)Comment to Reply 1: Quote:Not at all, almost killed myself /Quote - Maybe next time you should hire a professional

I'm just trying to get a handle on what you are asking here.  I believe that what I typed above is what you are asking about.  I don't see how this is possible given the current setup.  This would take some major rewriting of the code and not just a template change.  Not to mention its more like facebook or disqus than a forum.  Do you have some other forum software in mind that currently does this?

One of the things that I absolutely hate about sites that use disqus is that its really hard to follow.  Parts of a topic that have more replies usually make it above things chronologically and make reading posts in a timeline view really hard.  So anything like this would need to be an option rather than hard coded.  I don't see it being a template change because you would almost need new buttons for comment.  

I could see where this is possible but as I said someone would have to work out the mechanics.  Comment would be another option that would show up under whatever item you clicked comment on.  Quote would function as normal with a reply.   Something would need to be able to be turned on or off that would make the comment button disappear if you didn't want this functionality.  

I could see where this would be helpful in some instances as long as it was an option.

**Comment by gregclemens** [comment]
Ha.  I realized I was over commenting about comments.

But, yes you got what I was trying to describe.  Sounds like it's not doable.  I guess enough people aren't asking for it.  It just seems logical to me to be able to comment (reply to) on a comment.  Sometimes people ask questions in a comment and you want to see how people would answer that question without having to search through a long list of others who are just commenting on the original topic.  This feature is quite common in news articles.  The other nice thing about what I'm describing is Comments on comments would also be "collapsed" so you could quickly skim over those replies if you didn't care to read.  

But, thanks for your answer.  I understand, you get what you pay for.  I appreciate you all making this a robust open source tool.

**Comment by frozendnn** [comment]
You are talking about threaded view. I think when AF was still owned by AM, they were trying to do it. I saw a screenshot of a threaded view for AF. Nothing was released though.

**Comment by artfuld** [comment]
As the db is set up this is absolutely doable.
![image](https://cloud.githubusercontent.com/assets/6116748/9207669/3bc7c046-406f-11e5-9665-8a2e4bbd2f65.png)
Right now when you click the reply-button on a post it saves an int (ReplyToId) that I think should be the topic that you wanted to reply to. Doesn't seem to work but it shouldn't be a biggie to fix that.

The question is with the view, should the replies to anything but the thread post be hidden/minimized or should everything visible? 
And ofcourse, is it worth the time it will take to do this :).
